zetup.py

```diff
@@ -0,0 +1,159 @@
+
+    project.config['flex-fuse'] = flex_fuse_config
+
+
+@defer.inlineCallbacks
+def task_wait_all_projects_updated(project):
+    yield ziggy.tasks.wait_all_projects_updated(project)
+
+
+@defer.inlineCallbacks
+def task_declare_project_updated(project, zetup_path, old_zetup_md5):
+    yield ziggy.tasks.declare_project_updated(project, zetup_path, old_zetup_md5)
+
+
+@defer.inlineCallbacks
+def task_update_sources(project):
+    yield ziggy.tasks.update_sources(project,
+                                     repositories=project.config['flex-fuse']['repositories'],
+                                     base_path=project.config['flex-fuse']['base_path'])
+
+
+@defer.inlineCallbacks
+def task_load_snapshot(project, repo_merges=None):
+    yield ziggy.tasks.load_snapshot(project,
+                                    project_path=project.config['root_projects']['flex-fuse']['path'],
+                                    base_path=project.config['flex-fuse']['base_path'],
+                                    repo_merges=repo_merges)
+
+
+@defer.inlineCallbacks
+def task_take_snapshot(project, filter=None):
+    yield ziggy.tasks.take_snapshot(project,
+                                    repositories=project.config['flex-fuse']['repositories'].keys(),
+                                    project_path=project.config['root_projects']['flex-fuse']['path'],
+                                    base_path=project.config['flex-fuse']['base_path'],
+                                    filter=filter)
+
+
+@defer.inlineCallbacks
+def task_verify_zetup_unchanged(project):
+    yield ziggy.tasks.wait_workspace_updated(project,
+                                             project.config['flex-fuse']['zetup_path'],
+                                             project.config['flex-fuse']['zetup_md5'])
+
+
+@defer.inlineCallbacks
+def task_build_images(project, version, mirror):
+    """
+    Internal build function
+    """
+
+    project.logger.debug('Building', version=version)
+    cwd = project.config['flex-fuse']['flex_fuse_path']
+    cmd = 'make release MIRROR={0} IGUAZIO_VERSION={1}'.format(mirror, version)
+
+    project.logger.debug('Building a release candidate', cwd=cwd, cmd=cmd)
+    out, err, code = yield ziggy.shell.run(project.ctx, cmd, cwd=cwd)
+    if code != 0:
+        msg = 'Build execution failed'
+        project.logger.warn(msg, code=code, out=out, err=err)
+        raise RuntimeError(msg)
+
+    project.logger.debug('Build task is done', out=out)
+
+
+@defer.inlineCallbacks
+def task_push_docker_images(project, repository, tag, pushed_images_file_path):
+    """
+    Internal publish function
+    """
+
+    project.logger.debug('Publishing',
+                         repository=repository,
+                         tag=tag,
+                         pushed_images_file_path=pushed_images_file_path)
+
+    cwd = project.config['flex-fuse']['flex_fuse_path']
+
+    docker_image_name = 'iguaziodocker/flex-fuse:{0}'.format(tag)
```

Comment:
1. Remove iguaziodocker/ from name
2. There's not `docker login` here - is that ok?

Reply:
1. will be done asap
2. whoever trigger ziggy, would require to initiate docker login beforehand (zarcho for an instance)